### DISCUSSION TITLE
Been looking for an i3wm replacement that works on top of Gnome for pages, and here is where I'm stuck.

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [paperwm/PaperWM#721] [Q&A] [answered]
I've also posted here: https://www.reddit.com/r/gnome/comments/185akq0/is_there_a_tiling_window_manager_that_sits_on_top/

I've tried:
- [Forge](https://github.com/forge-ext/forge)
  - super buggy window resizing: https://github.com/forge-ext/forge/issues/305
- [PopShell](https://github.com/pop-os/shell)
  - breaks all the time
  - constant overlapping windows

What I want:
- all windows _wholly contained_ within the boundary of my screen
- spacial focus movement ( focus, window-movement :: left/right/up/down)
- ideally, modal-based resizing (because I've run out of ergonomic keys)
  - in the current window, move { left, right, top, bottom } edge { away, closer }

In trying PaperWM, I've found
- there is a similar left-boundary resize issue as Forge, where it makes the right side freak out (in the end, it's correct, but it's goofy).
- Doesn't seem like there's an option to keep windows viewable?
- I found that "Window Focus Mode" can surprise me, so I'd like to be able to disable the option to switch (not a big deal tho, as I've disabled 90% of the hotkeys)
- I disabled the minimap, because when using spatial-movement, you always know where you're going, so you don't need the minimap / alt-tab UI

Questions / gaps I see in PaperWM (I installed from https://extensions.gnome.org/extension/6099/paperwm/ )
- Can I disable the workspaces behavior? I just use the built stuff 
  - Workspace name
  - etc
- Is there a way to force all windows to be wholly contained within the monitor boundary and not have the scrolling behavior? I find it nauseating
  - Related, I don't really want gaps on either side of my windows either -- always use the available space
- I didn't see that this existed, but could a modal-based currently-focused-window resize mode be added?

Thanks!

**Comment by Rutherther** [comment]
What do you mean by keeping windows viewable? There is a scratch layer

**Reply by NullVoxPopuli** [reply]
> What do you mean by keeping windows viewable? 

windows keep sliding off the to the left or right, especially as I add more windows.

> There is a scratch layer

I don't want this, I don't think. It seems like a one-off thing, and I'm talking about my primary window workflow.

**Reply by tadfisher** [reply]
> all windows wholly contained within the boundary of my screen

PaperWM might not be what you're looking for, then. Its fundamental model is based on scrolling window strips on each of your desktops, and adding code to deal with different tiling algorithms kind of breaks the model.

I suggest [Material Shell](https://material-shell.com/) if you're looking for a more traditional tiling setup; it's fairly similar in scope to PaperWM.

**Comment by Lythenas** [comment, marked as answer]
PaperWM is definitively not a direct replacement for a normal tiling window manager.

The scrolling is kind of an integral part of the implementation and also a core concept of PaperWM. There is currently no way to disable the scrolling behavior.

Depending on how complicated that would make the code, I think we would accept PRs for it but I don't know if anyone would be willing to work on this themself. I also came from tiling window managers to PaperWM. It did take some getting used to in the beginning but it was definitively much better than plain Gnome.

If you find the scrolling nauseating you could try to set the animation speed to `0.01` (0.00 uses the default speed) so the animation is basically instant, but not sure how much that would help.

**Reply by NullVoxPopuli** [reply]
no worries, I think it's totally fine for a tool to not fit my use case!

thanks!!
